lib/formtastic/inputs/base/validations.rb

```diff
@@ -138,7 +138,11 @@ def required?
           if validations?
             validations.select { |validator|
               if validator.options.key?(:on)
-                return false if (validator.options[:on] != :save) && ((object.new_record? && validator.options[:on] != :create) || (!object.new_record? && validator.options[:on] != :update))
+                if validator.options[:on].is_a?(Array)
+                  return false if (validator.options[:on].exclude?(:save)) && ((object.new_record? && validator.options[:on].exclude?(:create)) || (!object.new_record? && validator.options[:on].exclude?(:update)))
```

Comment:
Could this be simplified by always converting it to an Array like `Array(validator.options[:on])` and then just checking for `exclude?` ?
Having similar conditions in two branches is really dangerous.